### PR TITLE
[native] Improve Zombie Tasks logging.

### PR DESCRIPTION
**Pull request by spershin**
## Description
1. Only get task info (toString) if really needed.
2. Log also the number of extra references.
3. Log the type of Task for each task.

If release note is NOT required, use:

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 1]
```diff
@@ -235,7 +236,9 @@ struct ZombieTaskStatsSet {
     tasks.reserve(numSampleTasks);
   }
 
-  void updateCounts(std::shared_ptr<exec::Task>& task) {
+  void updateCounts(
+      std::shared_ptr<exec::Task>& task,
+      const long numExtraReferences) {
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:241]: drop const here as it is scalar function input. Thanks!

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 2]
```diff
@@ -207,15 +207,16 @@ void checkSplitsForBatchTask(
...
-
-  std::string toString() const {
-    return SystemConfig::instance()->logZombieTaskInfo() ? taskInfo : taskId;
-  }
+  const std::string info;
+  const long numExtraReferences;
+
+  ZombieTaskStats(
+      const std::shared_ptr<exec::Task>& task,
+      long numExtraReferences_)
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:215]: Comment say the extra references could be either from velox or presto? Thanks!

**Review by xiaoxmeng — DISMISSED** [review]
@spershin thanks for the improvement % nits.

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 3]
```diff
@@ -207,15 +207,16 @@ void checkSplitsForBatchTask(
...
-  explicit ZombieTaskStats(const std::shared_ptr<exec::Task>& task)
-      : taskId(task->taskId()), taskInfo(task->toString()) {}
-
-  std::string toString() const {
-    return SystemConfig::instance()->logZombieTaskInfo() ? taskInfo : taskId;
-  }
+  const std::string info;
+  const long numExtraReferences;
+
+  ZombieTaskStats(
```
Comment (czentgr) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:213]: Nit: Why was `explicit` removed? This should still be used for the construction during the emplace call like before?

File: presto-native-execution/presto_cpp/main/TaskManager.cpp (hunk 2 above)
Comment (spershin) [reply presto-native-execution/presto_cpp/main/TaskManager.cpp:215]: This is conveyed on another level.
The structure has no knowledge about it.

File: presto-native-execution/presto_cpp/main/TaskManager.cpp (hunk 3 above)
Comment (spershin) [reply presto-native-execution/presto_cpp/main/TaskManager.cpp:213]: explicit is was used for single-argument constructors, so there would be no implicit type conversion.

**Review by czentgr — DISMISSED** [review]

File: presto-native-execution/presto_cpp/main/TaskManager.cpp [hunk 4]
```diff
@@ -207,15 +207,16 @@ void checkSplitsForBatchTask(
...
+      const std::shared_ptr<exec::Task>& task,
+      long _numExtraReferences)
```
Comment (xiaoxmeng) [inline presto-native-execution/presto_cpp/main/TaskManager.cpp:215]:
```
s/long/uint32_t/ why long? Thanks!
```

**Review by xiaoxmeng — APPROVED** [review]

File: presto-native-execution/presto_cpp/main/TaskManager.cpp (hunk 4 above)
Comment (spershin) [reply presto-native-execution/presto_cpp/main/TaskManager.cpp:215]: @xiaoxmeng
Simply based on where we get them from:
`const auto prestoTaskRefCount = prestoTask.use_count();`
which comes from
`long use_count() const _NOEXCEPT`

**Pull request merged**